java/fory-core/src/main/java/org/apache/fory/builder/BaseObjectCodecBuilder.java

```diff
@@ -405,7 +405,17 @@ protected Expression serializeField(
     TypeRef<?> typeRef = descriptor.getTypeRef();
     boolean nullable = descriptor.isNullable();
 
+    // ref tracking logic:
+    // - if the ref tracking is enabled globally, then @ForyField will be checked
+    // - otherwise, fallback to global config
+    boolean useRefTracking;
     if (needWriteRef(typeRef)) {
+      useRefTracking = descriptor.getForyField() == null || descriptor.isTrackingRef();
```

Comment:
with https://github.com/apache/fory/pull/3021/files#r2625961552, we can just use `descriptor.isTrackingRef()`